File: client/src/types/episode.type.ts
import { PartialData } from "./index"
import { Block } from "./block.type"

export interface PartialEpisode extends PartialData {
  title: string
  description: string
  chapter: string
  novelId: string
  order: number
  episodeType: EpisodeType
  createdAt: Date
  updatedAt: Date
}

export interface Episode extends PartialEpisode {
  novel: PartialData
  blocks: Block[]
}

export enum EpisodeType {
  Episode,
  EpisodeGroup,
  Prologue,
  Epilogue,
  Special,
}

export const initialPartialEpisode: PartialEpisode = {
  id: "",
  title: "",
  chapter: "",
  description: "",
  novelId: "",
  order: 0,
  episodeType: EpisodeType.Episode,
  createdAt: new Date(),
  updatedAt: new Date(),
}

File: packages/tauri-app/src-tauri/src/models/enums/episode_type.rs
use serde_repr::{Deserialize_repr, Serialize_repr};

#[derive(Serialize_repr, Deserialize_repr, Debug, Clone, PartialEq)]
#[repr(u8)]
pub enum EpisodeType {
    Episode = 0,
    EpisodeGroup = 1,
    Prologue = 2,
    Epilogue = 3,
    Special = 4,
    Memo = 5,
}
